### PR TITLE
Add create, delete and configure actions to windows_service

### PR DESCRIPTION
**Pull request by jasonwbarnett**
### Description

adds `:create`, `:delete`, and `:configure` actions to the `windows_service` resource.

### Issues Resolved

- #2518
- #6595
- https://tickets.opscode.com/browse/CHEF-3865#

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by jasonwbarnett** [comment]
@btm @lamont-granquist @thommay -- I finally added tests and ensured they all pass! :blush:

**Comment by tas50** [comment]
@jasonwbarnett You have some chefstyle warnings. Run chefstyle -a . on the repo and see if there's anything that can't be autocorrected

**Comment by jasonwbarnett** [comment]
@tas50 Just added an additional commit to get this to pass chefstyle

File: lib/chef/resource/windows_service.rb [hunk 1]
```diff
@@ -17,58 +17,101 @@
...
+      # services or load ordering groups that the system must start before this
+      # service. Specify nil or an empty string if the service has no
+      # dependencies. Dependency on a group means that this service can run if
+      # at least one member of the group is running after an attempt to start
+      # all members of the group.
+      property :dependencies, [String, Array]
+
+      property :description, String
+
+      property :run_as_user, String, default: "LocalSystem"
```
Comment (jasonwbarnett) [inline lib/chef/resource/windows_service.rb]: I want to point out that I've updated the default of `run_as_user` from `""` to `"LocalSystem"`

**Comment by lamont-granquist** [comment]
looks like on non-windows the unit tests are failing hard:

```
An error occurred while loading ./spec/unit/provider/service/windows_spec.rb.
Failure/Error: class Chef::ReservedNames::Win32::Security; end
TypeError:
  Security is not a class
# ./spec/unit/provider/service/windows_spec.rb:23:in `<top (required)>'
```

**Comment by jasonwbarnett** [comment]
I had added a line to the spec file to get the tests to run on MacOS (still not sure how to run these unit tests on a non-windows OS) -- The removal of that line should fix it.

**Comment by jasonwbarnett** [comment]
@lamont-granquist Can you show me where/how you are seeing these errors so I can also?

**Comment by lamont-granquist** [comment]
yeah our CI tests are a bit of a mess -- travis often fails on the unit tests and throws a bunch of EEXIST errors on stuff which is something that we've been debugging with travis for months now (it only happens on some of their tester boxes)  then there's issues with poise and halite, and the odd timeout error that happens sometimes as well.  appveyor also has powershell_out failures that haven't been fixed yet.

**Comment by lamont-granquist** [comment]
yeah this looks as green as it gets right now

**Comment by jasonwbarnett** [comment]
I can’t tell if your saying that the powershell_out failures are related to this pull or just in general.

Is this pull look good? Anything you see missing or that should be re-worked? Different design? Etc

**Comment by thommay** [comment]
Thanks for taking the time to make this better @jasonwbarnett

**Pull request merged**